tasks/pkg-suse.yml

```diff
@@ -0,0 +1,62 @@
+---
+- name: Stat if RPM key already exist
+  stat:
+    path: /tmp/DATADOG_RPM_KEY.public
+  register: ddkey
+
+- name: Download new RPM key
+  shell: "curl {{ datadog_zypper_gpgkey }} -o /tmp/DATADOG_RPM_KEY.public" # Work around due to SNI check for SLES11
+  args:
+    warn: no
+  when: not ddkey.stat.exists
+
+- name: Import new RPM key
+  rpm_key:
+    key: /tmp/DATADOG_RPM_KEY.public
+    state: present
+  when: not ansible_check_mode
+
+- name: Install DataDog zypper repo
+  zypper_repository:
+    name: datadog
+    description: Datadog, Inc.
+    repo: "{{ datadog_yum_repo }}"
+    enabled: yes
+    auto_import_keys: yes
+    disable_gpg_check: yes # SLES11 considers datadog's repo as untrusted
+    state: "{% if datadog_agent5 %}absent{% else %}present{% endif %}"
+
+- name: Install DataDog yum repo (agent5)
+  zypper_repository:
+    name: datadog-beta
+    description: Datadog, Inc.
+    repo: "{{ datadog_agent5_yum_repo }}"
+    enabled: yes
+    auto_import_keys: yes
+    disable_gpg_check: yes # SLES11 considers datadog's repo as untrusted
+    state: "{% if datadog_agent5 %}present{% else %}absent{% endif %}"
+
+- name: Drop SysVinit script if SLES 11
```

Comment:
We do not support sysvinit for agent6.  The agent run on Suse11 if you're running systemd.

We don't want to support any sysvinit script, especially in our ansible repo. If users want to use sysvinit they will have to ship and maintain their own script.